From 243f6ead0f388afb0a010d5524bdafd421848acc Mon Sep 17 00:00:00 2001
From: Alex Zhao <zzc@rock-chips.com>
Date: Mon, 14 May 2018 11:24:46 +0800
Subject: [PATCH] net: wireless: rockchip_wlan: fix ap6354 sleep issue

do not let the kernel go to sleep while TCP packets are being sent.

Change-Id: I31b86af06f97816d28e4ad691ac6d37973724329
Signed-off-by: Alex Zhao <zzc@rock-chips.com>
---
 .../net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_config.c | 2 ++
 .../net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c  | 8 +++++++-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_config.c b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_config.c
index f6d30ba7a920..47480b261dd1 100644
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_config.c
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_config.c
@@ -2623,6 +2623,8 @@ dhd_conf_preinit(dhd_pub_t *dhd)
 		conf->txglomsize = SDPCM_MAXGLOM_SIZE;
 	conf->deferred_tx_len = conf->txglomsize;
 #endif
+	if (conf->chip == BCM4354_CHIP_ID && conf->chiprev == 1)
+		dhd_slpauto = 0;
 
 	return 0;
 }
diff --git a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c
index 2f9078fba777..ef17fe10d0d6 100644
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c
@@ -5029,6 +5029,7 @@ dhd_start_xmit(struct sk_buff *skb, struct net_device *net)
 #endif /* DHD_PCIE_RUNTIMEPM */
 
 	DHD_GENERAL_LOCK(&dhd->pub, flags);
+#ifdef BCMPCIE
 	if (DHD_BUS_CHECK_SUSPEND_OR_SUSPEND_IN_PROGRESS(&dhd->pub)) {
 		DHD_ERROR(("%s: bus is in suspend(%d) or suspending(0x%x) state!!\n",
 			__FUNCTION__, dhd->pub.busstate, dhd->pub.dhd_bus_busy_state));
@@ -5047,7 +5048,12 @@ dhd_start_xmit(struct sk_buff *skb, struct net_device *net)
 		return NETDEV_TX_BUSY;
 #endif
 	}
-
+#else
+	if (DHD_BUS_CHECK_SUSPEND_OR_SUSPEND_IN_PROGRESS(&dhd->pub)) {
+		DHD_ERROR(("%s: bus is in suspend(%d) or suspending(0x%x) state!!\n",
+			__FUNCTION__, dhd->pub.busstate, dhd->pub.dhd_bus_busy_state));
+	}
+#endif
 	DHD_OS_WAKE_LOCK(&dhd->pub);
 	DHD_PERIM_LOCK_TRY(DHD_FWDER_UNIT(dhd), lock_taken);
 
-- 
2.35.3

